packages/gatsby-theme-newrelic/src/components/SearchInput.js

```diff
@@ -0,0 +1,150 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { css } from '@emotion/core';
+import styled from '@emotion/styled';
+import Icon from './Icon';
+
+const SIZES = {
+  MEDIUM: 'medium',
+  LARGE: 'large',
+};
+
+const SearchInput = ({
+  onClear,
+  value,
+  width,
+  size = 'medium',
+  className,
+  style,
+  ...props
+}) => {
+  const handleClick = (e) => {
+    e.preventDefault();
+    onClear();
+  };
+  const handleKeyDown = (e) => {
+    if (e.key === 'Escape') onClear();
+  };
+
+  return (
+    <StyledContainer
+      width={width}
+      className={className}
+      style={style}
+      size={size}
+    >
+      <StyledIcon
+        css={css`
+          position: absolute;
+          left: 1rem;
+          top: 50%;
+          transform: translateY(-50%);
+        `}
+        name={Icon.TYPE.SEARCH}
+        size={size}
+      />
+      <StyledInput
+        value={value}
+        size={size}
+        {...props}
+        type="text"
+        onKeyDown={handleKeyDown}
+      />
+      {value && (
+        <StyledButton
+          onClick={handleClick}
+          css={css`
+            right: 1rem;
+            top: 50%;
+            transform: translateY(-50%);
+            &:hover {
+              cursor: pointer;
+            }
+          `}
+          onKeyDown={(e) => e.preventDefault()}
```

Comment:
[nit] It looks like you're mixing styles a bit with how you pass the function to the event handlers. I see the `onClick` passes a function declared from a variable where this one is declared inline.

I personally prefer the inline style like you have here since it means I don't need to go search for the variable to understand what the handler does. Regardless, could you try and unify the styles a bit and either go all handlers inline or all passed from variables?